### ISSUE TITLE
Make logging impl pluggable

### ISSUE THREAD

**Issue by robvdlv** [JustGiving/JustGiving.EventStore.Http#23]
The lib now takes an explicit Log4Net dependency.
Library users should be able to use logging of their liking.

Some options:

- introduce a minimal logging api in the library so lib users can plug their own lightweight adapter; this is 
- use LibLog (https://github.com/damianh/LibLog) to transparently bind to commonly used specific logging lib impls

What do you guys think?

**Comment by spadger** [comment]
Yeah, i'm not against it, seeing how small the dependency is, given the fact that it wouldn't break the api

**Comment by robvdlv** [comment]
Any option that takes your preference? Or alternatively you may have another?

**Comment by spadger** [comment]
Hmm,


LibLog seems to be the way forward, but I'd like to get by without a separate dependency.  I also need to think about how to let users choose which logging framework  to use if they have referenced multiple.  This may be just a case of them configuring liblog directly, but I'd like to play first.


Jon


________________________________
